Tests fail on Node 6

Node's net library has a breaking change in Node 6, which performs validation on the port in the listen method.

```
~/Repos/koala → npm test

> koala@1.0.0 test /Users/nick/Repos/koala
> NODE_ENV=test mocha --reporter spec --timeout 30s --bail test/app/index.js

internal/net.js:17
    throw new RangeError('"port" argument must be >= 0 and < 65536');
    ^

RangeError: "port" argument must be >= 0 and < 65536
    at assertPort (internal/net.js:17:11)
    at Server.listen (net.js:1384:5)
    at Application.listen (/Users/nick/Repos/koala/lib/app.js:19:12)
    at Suite.<anonymous> (/Users/nick/Repos/koala/test/app/conditional-get.js:11:22)
    at context.describe.context.context (/Users/nick/Repos/koala/node_modules/mocha/lib/interfaces/bdd.js:47:10)
    at Suite.<anonymous> (/Users/nick/Repos/koala/test/app/conditional-get.js:3:3)
    at context.describe.context.context (/Users/nick/Repos/koala/node_modules/mocha/lib/interfaces/bdd.js:47:10)
    at Object.<anonymous> (/Users/nick/Repos/koala/test/app/conditional-get.js:2:1)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
npm ERR! Test failed.  See above for more details.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,7 @@
 language: node_js
 node_js:
-  - 1
-  - 2
-  - 3
   - 4
-  - 5
   - 6
-  - 7
   - 8
 sudo: false
 script: "npm run test-travis"

--- a/lib/app.js
+++ b/lib/app.js
@@ -16,7 +16,7 @@ module.exports = function (app) {
       req.checkContinue = true;
       fn(req, res);
     });
-    server.listen(port || process.env.PORT, cb);
+    server.listen(port || process.env.PORT || 0, cb);
     return server;
   }
 }

